### PR TITLE
Create 2020-06-23-SE1.md

### PR DESCRIPTION
**Pull request by JeanMarieMineau**
Article about the collision function.

### CODE REVIEW

File: _posts/2020-06-23-SE1.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+tags:
+ - update
+ - progress
+ - software engineering
+ - collision
+---
+
+н3110 ω0Я1δ,
+
+This morning (well this morning, at 3 am...) I finished the a first function (https://github.com/PainsPerdus/gboi-rocket/wiki/collision-doc). The collision function is a critical function of the game, which will be called a lot of time. Thus, this function is meant to evolve.
```
Comment (DorianXGH) [inline _posts/2020-06-23-SE1.md:16]: the ~a~ first function
lot of timeS

Comment (DorianXGH) [inline _posts/2020-06-23-SE1.md:16]: I finished the ~a~ first function
a lot of timeS

File: _posts/2020-06-23-SE1.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+We had debates yesterday and this morning (the real morning this time) about what this function should do, or what argument we should give it. For now this version takes a position and a size id for both element checked. We may implement an other function for the bullets which takes only a point for the first element. Another optimization possible is to tell if the first hitbox is already loaded.
```
Comment (DorianXGH) [inline _posts/2020-06-23-SE1.md:18]: We ~had debates~ debated
another, not an other

File: _posts/2020-06-23-SE1.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+Dorian pointed out to me that using the RAM to store the variable is SLOOOOOOOWWWWWWW. Too bad for a critical function. The thing is that this function needs (at least) 8 variables (and even a little more), so there are not enough registers. We are thinking of using the HRAM as a cache for functions, and putting the stack in the WRAM.
```
Comment (DorianXGH) [inline _posts/2020-06-23-SE1.md:20]: Turns out I was wrong and HRAM and WRAM are the same speed, sorry :/

File: _posts/2020-06-23-SE1.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+For now, this function is tested with pong game, and it works well. We will use it for the first milestone. Then we will consider our options to optimize this function if needed.
```
Comment (DorianXGH) [inline _posts/2020-06-23-SE1.md:22]: with THE pong game

File: _posts/2020-06-23-SE1.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+I also used this function think about the documentation and used it for the first documentation page. (More about that in my next article :smirk_cat:)
```
Comment (DorianXGH) [inline _posts/2020-06-23-SE1.md:24]: used this function TO think

**Comment by g33kex** [comment]
I recommend using Grammarly to fix your mistakes, it works well

**Pull request merged**